sympy/printing/str.py

```diff
@@ -981,19 +981,55 @@ def _print_AppliedBinaryRelation(self, expr):
 
 @print_function(StrPrinter)
 def sstr(expr, **settings):
-    """Returns the expression as a string.
-
-    For large expressions where speed is a concern, use the setting
-    order='none'. If abbrev=True setting is used then units are printed in
-    abbreviated form.
+    """
+    Returns the expression *expr* as a string.
+
+    Parameters
+    ==========
+    expr : :class:`.Expr`
+        Expression to be printed.
+    order : str or None, default: None
+        Any of the supported monomial orderings (currently 'lex',
+        'grlex', or 'grevlex'), 'old', and 'none'. This parameter does nothing
+        for :class:`.Mul` objects. Setting order to 'old' uses the
+        compatibility ordering for :class:`.Add` defined in :class:`.Printer`.
+        For very large expressions, set the *order* keyword to 'none' if
+        speed is a concern.
+    full_prec : bool or 'auto', default: 'auto'
+        Whether to print trailing zeros for :class:`.Float`. If 'auto', print
+        full precision for individual floats, but not when part of an
+        expression.
+    sympy_integers : bool, default: False
+        If True, an integer is printed surrounded by a call to S,
+        to get a SymPy :class:`.Integer` upon evaluation of the string.
+    abbrev : bool, default: False
+        Whether to print units in abbreviated form.
+    perm_cyclic : bool, default: True
+        See :class:`.Permutation` for info.
+    min : int or None, default: None
+    max : int or None, default: None
```

Comment:
It would be good to say something about what these do here rather than just the examples below. Probably these two parameters should be grouped with `full_prec` since those three together control the printing of floats.

Reply:
Yeah, I was trying to figure out what they did but couldn't... And they are not documented in mpmath either.

Regarding ordering this is the same order as in the default-dict and therefore the doc-string. Now, these are keyword only, but I am still not sure it is a good idea to mix orders (although your point of grouping is also valid).

Reply:
We might consider renaming these parameters to something more meaningful.

Reply:
More about min/max here in libmpf:
```
def to_str(s, dps, strip_zeros=True, min_fixed=None, max_fixed=None,
    show_zero_exponent=False):
    """
    Convert a raw mpf to a decimal floating-point literal with at
    most `dps` decimal digits in the mantissa (not counting extra zeros
    that may be inserted for visual purposes).
    The number will be printed in fixed-point format if the position
    of the leading digit is strictly between min_fixed
    (default = min(-dps/3,-5)) and max_fixed (default = dps).
    To force fixed-point format always, set min_fixed = -inf,
    max_fixed = +inf. To force floating-point format, set
    min_fixed >= max_fixed.
```
From limited testing I think this means that if the exponent, e, is min < e < max then floating format will be used, else exponential:

![image](https://user-images.githubusercontent.com/90703/198816770-06b18d83-2a78-4bee-b95b-eff6e8c32a8b.png)